types/request/index.d.ts

```diff
@@ -227,7 +227,7 @@ declare namespace request {
         json(val: any): Request;
         aws(opts: AWSOptions, now?: boolean): Request;
         hawk(opts: HawkOptions): void;
-        auth(username: string, password: string, sendImmediately?: boolean, bearer?: string): Request;
+        auth(username: string | null, password: string | null, sendImmediately?: boolean, bearer?: string): Request;
```

Comment:
Does it make more sense to overload this method with the two forms?
```ts
auth(username: string, password: string, sendImmediately?: boolean): Request;
auth(username: null, password: null, sendImmediately: boolean, bearer: string): Request;
```
This would make a call like `...auth('bob', 'password', true, 'sometoken')` show an error.
The [function def is here](https://github.com/request/request/blob/master/request.js#L1338) which just proxies to [this function](https://github.com/request/request/blob/master/lib/auth.js#L126).

Reply:
Totally! Yeah, that makes a lot more sense. One definitely wouldn't want to send both a username:pass and a token at the same time.

Reply:
@nwehner: Do you plan to make this change or should I merge this PR?

Reply:
Just edited the typings to include both options for 'auth'